src/block/BaseRail.php

```diff
@@ -220,11 +222,13 @@ private function setConnections(array $connections) : void{
 	}
 
 	public function onNearbyBlockChange() : void{
-		if($this->getSide(Facing::DOWN)->isTransparent()){
+		$supportType = $this->getSide(Facing::DOWN)->getSupportType(Facing::UP);
+		if($supportType->equals(SupportType::NONE()) || $supportType->equals(SupportType::CENTER())){
 			$this->position->getWorld()->useBreakOn($this->position);
 		}else{
 			foreach($this->getCurrentShapeConnections() as $connection){
-				if(($connection & RailConnectionInfo::FLAG_ASCEND) !== 0 && $this->getSide($connection & ~RailConnectionInfo::FLAG_ASCEND)->isTransparent()){
+				$supportType = $this->getSide($connection & ~RailConnectionInfo::FLAG_ASCEND)->getSupportType(Facing::UP);
+				if(($connection & RailConnectionInfo::FLAG_ASCEND) !== 0 && ($supportType->equals(SupportType::NONE()) || $supportType->equals(SupportType::CENTER()))){
```

Comment:
I'm not convinced this is correct. Slope support should depend on the face of the block it leans on, not the top face.

Reply:
`!supportType->hasEdgeSupport()` can be used here

Reply:
It is not even either of the two logics. As can be tested in vanilla, by putting a hopper and rails, the rail seems to be based on the block below it all the time.